src/components/ViewedByCustomer/ViewedByCustomer.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, ComponentOptions, IComponentBindings, Initialization, l } from 'coveo-search-ui';
+import { user } from '../../utils/icons';
+import './Strings';
+
+/**
+ * The options for the ViewedByCustomerComponent
+ */
+export interface IViewedByCustomerOptions {
+    /**
+     * If true, will display an icon when the component is displayed itself.
+     */
+    showIcon?: boolean;
+    /**
+     * The label that should be displayed when the component is displayed.
+     */
+    label?: string;
+}
+
+/**
+ * Result component indicating if a search result have been viewed by the targeted user.
+ */
+export class ViewedByCustomer extends Component {
+    /**
+     * Unique Identifier used by the Search-UI.
+     */
+    public static readonly ID = 'ViewedByCustomer';
+
+    /**
+     * Default options used by the component.
+     */
+    public static readonly options: IViewedByCustomerOptions = {
+        showIcon: ComponentOptions.buildBooleanOption({ defaultValue: true }),
```

Comment:
I know you just moved the comments to the interface, but if I remember right, the comments need to be on the static options for the interface editor to display them.

Reply:
The interface editor is not supported (Unless they have been massive simplification) in this PR, so I do not think we should bother with this question right away, because we'll be unable to verify.